### PR TITLE
Add support of contextual quick info

### LINKED ISSUE #20966: get (or set) function shows description from *both* get and set function on hover
_From @kleinfreund on December 24, 2017 9:36_

- VSCode Version: Code 1.19.1 (0759f77bb8d86658bc935a10a64f6182c5a1eeba, 2017-12-19T09:41:01.414Z)
- OS Version: Linux x64 4.13.0-21-generic
- Extensions: Extensions are disabled
---

Steps to Reproduce:

1. View the following code with language mode set to JavaScript

   ```js
   class Test {
     constructor() {
       this.value;
     }
   
     /** Getter text */
     get value() {
       return this.value;
     }
   
     /** Setter text */
     set value(value) {
       this.value = value;
     }
   }
   ```

2. Hover over the `value` getter (or setter) function

**Expected behavior**:

`Getter text` (or `Setter text`) appears in the box as the function description.

**Actual behavior**:

`Getter text Setter text` appears in the box as the function description.

_Copied from original issue: Microsoft/vscode#40775_

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #20966

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -283,6 +283,12 @@ namespace ts {
         }
     }
 
+    enum ContextualDocumentationType {
+        None = "None",
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Nit: we prefer numeric const enums for efficiency

Comment (Kingwl) [reply src/services/services.ts]: Seems `Map` interface cannot work with the numerical enum, does it?

Comment (sheetalkamat) [reply src/services/services.ts]: It doesn't need to be map.. Jus two properties that store array of decl comments one for get and one for set accessor ?

Comment (Kingwl) [reply src/services/services.ts]: Okay...but I felt maybe it will take more contextual info in the future. For example getter or setter in LHS or RHS

Comment (Kingwl) [reply src/services/services.ts]: But never mind, We could refactor in time.

**Comment by Kingwl** [comment]
up 🆙

**Pull request merged**